addons/knowledge/models/knowledge_article.py

```diff
@@ -162,6 +170,18 @@ def _check_parent_id_recursion(self):
                  )
             )
 
+    @api.constrains('child_ids')
+    def _check_article_item_children(self):
+        for article in self:
+            if article.item_child_ids and article.article_child_ids:
```

Comment:
``if len(set(article.child_ids.mapped('is_article_item'))) > 1`` ?

Reply:
no because child_ids can contain both. so I need to check if there are 1 article of each type.